tasklist/client/src/Tasks/Task/Variables/VariableEditor/index.tsx

```diff
@@ -0,0 +1,314 @@
+                  name={createVariableFieldName(variable.name)}
+                  validate={
+                    variable.isValueTruncated
+                      ? () => undefined
+                      : validateValueJSON
+                  }
+                >
+                  {({input, meta}) => (
+                    <LoadingTextarea
+                      {...input}
+                      id={input.name}
+                      invalidText={meta.error}
+                      isLoading={variablesLoadingFullValue.includes(
+                        variable.id,
+                      )}
+                      onFocus={(event) => {
+                        if (variable.isValueTruncated) {
+                          fetchFullVariable(variable.id);
+                        }
+                        input.onFocus(event);
+                      }}
+                      isActive={meta.active}
+                      type="text"
+                      labelText={`${variable.name} value`}
+                      placeholder={`${variable.name} value`}
+                      hideLabel
+                    />
+                  )}
+                </Field>
+              </StructuredListCell>
+              <StructuredListCell
+                className={cn(styles.listCell, styles.controlsCell)}
+              >
+                <div className={cn(styles.iconButtons, styles.extraPadding)}>
+                  <IconButton
+                    label={CODE_EDITOR_BUTTON_TOOLTIP_LABEL}
+                    onClick={() => {
+                      if (variable.isValueTruncated) {
+                        fetchFullVariable(variable.id);
+                      }
+
+                      onEdit(createVariableFieldName(variable.name));
+                    }}
+                    size="sm"
+                    kind="ghost"
+                    leaveDelayMs={100}
+                  >
+                    <Popup />
+                  </IconButton>
+                </div>
+              </StructuredListCell>
+            </StructuredListRow>
+          ) : (
+            <StructuredListRow key={variable.name}>
+              <StructuredListCell
+                className={cn(styles.listCell, styles.cellName)}
+              >
+                {variable.name}
+              </StructuredListCell>
+              <StructuredListCell
+                className={cn(styles.listCell, styles.valueCell)}
+              >
+                <div className={styles.scrollableOuter}>
+                  <div className={styles.scrollableInner}>
+                    {variable.isValueTruncated
+                      ? `${variable.previewValue}...`
+                      : variable.value}
+                  </div>
+                </div>
+              </StructuredListCell>
+              <StructuredListCell
+                className={cn(styles.listCell, styles.controlsCell)}
+              />
+            </StructuredListRow>
+          ),
+        )}
+        {!readOnly ? (
```

Comment:
Let's invest the logic to avoid unnecessary nots

```suggestion
        {readOnly ? (
```

Reply:
Still stands